crates/fluvio-types/src/compression.rs

```diff
@@ -0,0 +1,79 @@
+use std::str::FromStr;
+
+use serde::{Serialize, Deserialize};
+
+#[derive(thiserror::Error, Debug)]
+pub enum CompressionParseError {
+    #[error("unknown compression format: {0}")]
+    UnknownCompressionFormat(String),
+}
+
+pub const COMPRESSION_CODE_NONE: i8 = 0;
+pub const COMPRESSION_CODE_GZIP: i8 = 1;
+pub const COMPRESSION_CODE_SNAPPY: i8 = 2;
+pub const COMPRESSION_CODE_LZ4: i8 = 3;
+pub const COMPRESSION_CODE_ZSTD: i8 = 4;
+
+pub const COMPRESSION_NONE: &str = "none";
+pub const COMPRESSION_GZIP: &str = "gzip";
+pub const COMPRESSION_SNAPPY: &str = "snappy";
+pub const COMPRESSION_LZ4: &str = "lz4";
+pub const COMPRESSION_ZSTD: &str = "zstd";
+
+/// The compression algorithm used to compress and decompress records in fluvio batches
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Hash)]
+#[serde(rename_all = "lowercase")]
+#[repr(i8)]
+#[derive(Default)]
+pub enum Compression {
```

Comment:
Seems like this shouldn't belong here.  This is compression coding.

Reply:
I thought this was what @digikata suggested this morning in slack.

We need a way for crates to include these types without including the compression algorithms.

This PR first attempted to accomplished this with the `types` feature on `fluvio-compression`. 

Then I thought we agreed to instead move `enum Compression` to `fluvio-types`.